src/stores.rs

```diff
@@ -335,23 +357,56 @@ impl<K: Hash + Eq, V> SizedCache<K, V> {
         self.order.iter().map(|(_k, v)| v)
     }
 
+    fn hash(&self, key: &K) -> u64 {
+        let hasher = &mut self.hash_builder.build_hasher();
+        key.hash(hasher);
+        hasher.finish()
+    }
+
+    fn insert_index(&mut self, hash: u64, index: usize) {
+        let Self {
+            ref mut store,
+            ref order,
+            ref hash_builder,
+            ..
+        } = *self;
+        store.insert(hash, index, move |&i| {
+            let hasher = &mut hash_builder.build_hasher();
+            order.get(i).0.hash(hasher);
+            hasher.finish()
+        });
+    }
+
+    fn get_index(&self, hash: u64, key: &K) -> Option<usize> {
+        let Self { store, order, .. } = self;
+        store.get(hash, |&i| *key == order.get(i).0).copied()
+    }
+
+    fn remove_index(&mut self, hash: u64, key: &K) -> Option<usize> {
+        let Self { store, order, .. } = self;
+        store.remove_entry(hash, |&i| *key == order.get(i).0)
+    }
+
     fn check_capacity(&mut self) {
         if self.store.len() >= self.capacity {
             // store has reached capacity, evict the oldest item.
             // store capacity cannot be zero, so there must be content in `self.order`.
-            let (key, _value) = self.order.pop_back();
-            self.store
-                .remove(&key)
-                .expect("SizedCache::cache_set failed evicting cache key");
+            let index = self.order.back();
+            let key = &self.order.get(index).0;
+            let hash = self.hash(key);
+
+            let order = &self.order;
+            let erased = self.store.erase_entry(hash, |&i| *key == order.get(i).0);
+            assert!(erased, "SizedCache::cache_set failed evicting cache key");
+            self.order.remove(index);
         }
     }
 }
 
-impl<K: Hash + Eq + Clone, V> Cached<K, V> for SizedCache<K, V> {
+impl<K: Hash + Eq, V> Cached<K, V> for SizedCache<K, V> {
```

Comment:
Note that `K: Clone` is no longer required.
(Although it might be desirable to keep that for API stability.)